lib/__tests__/ModuleFinder-test.js

```diff
@@ -0,0 +1,94 @@
+import path from 'path';
+
+import ModuleFinder from '../ModuleFinder';
+import findPackageDependencies from '../findPackageDependencies';
+import normalizePath from '../normalizePath';
+import readFile from '../readFile';
+import requireResolve from '../requireResolve';
+
+jest.mock('../readFile');
+jest.mock('../findPackageDependencies');
+jest.mock('../requireResolve');
+
+let moduleFinder;
+
+function addFiles(files) {
+  return moduleFinder.handleFilesAdded(files.map((file) => ({
+    path: normalizePath(file, process.cwd()),
+    mtime: Date.now(),
+  })));
+}
+
+beforeEach(() => {
+  readFile.mockImplementation(() => Promise.resolve('export default {}'));
+
+  findPackageDependencies.mockImplementation(() => ['pacman', 'react-packer']);
+
+  requireResolve.__addResolvedPath(
+    path.join(process.cwd(), 'node_modules/pacman'),
+    path.join(process.cwd(), 'node_modules/pacman/index.js'),
+  );
+
+  requireResolve.__addResolvedPath(
+    path.join(process.cwd(), 'node_modules/react-packer'),
+    path.join(process.cwd(), 'node_modules/react-packer/index.js'),
+  );
+
+  moduleFinder = ModuleFinder.getForWorkingDirectory(process.cwd(), {
+    excludes: [],
+    ignorePackagePrefixes: ['react-'],
+  });
+  return moduleFinder.initializeStorage(':memory:');
```

Comment:
I agree with you in general, but I also think it's fine in this one case. The `Importer` test is an example of where this has gone way out of proportion (sigh).